internal/iapl/rbac.go

```diff
@@ -0,0 +1,214 @@
+package iapl
+
+import (
+	"go.infratographer.com/permissions-api/internal/types"
+)
+
+const (
+	// RoleOwnerRelation is the name of the relationship that connects a role to its owner.
+	RoleOwnerRelation = "owner"
+	// RoleOwnerMemberRoleRelation is the name of the relationship that connects a resource
+	// to a role that it owns
+	RoleOwnerMemberRoleRelation = "member_role"
+	// AvailableRoleRelation is the name of the relationship that list all roles
+	// that are available for a given resource
+	AvailableRoleRelation = "avail_role"
+	// RolebindingRoleRelation is the name of the relationship that connects a role binding to a role.
+	RolebindingRoleRelation = "role"
+	// RolebindingSubjectRelation is the name of the relationship that connects a role binding to a subject.
+	RolebindingSubjectRelation = "subject"
+	// RoleOwnerParentRelation is the name of the relationship that connects a role's owner to its parent.
+	RoleOwnerParentRelation = "parent"
+	// PermissionRelationSuffix is the suffix append to the name of the relationship
+	// representing a permission in a role
+	PermissionRelationSuffix = "_rel"
+	// GrantRelationship is the name of the relationship that connects a role binding to a resource.
+	GrantRelationship = "grant"
+)
+
+// RoleBindingAction is the list of actions that can be performed on a role-binding resource
+type RoleBindingAction string
+
+const (
+	// RoleBindingActionCreate is the action name to create a role binding
+	RoleBindingActionCreate RoleBindingAction = "rolebinding_create"
+	// RoleBindingActionUpdate is the action name to update a role binding
+	RoleBindingActionUpdate RoleBindingAction = "rolebinding_update"
+	// RoleBindingActionDelete is the action name to delete a role binding
+	RoleBindingActionDelete RoleBindingAction = "rolebinding_delete"
+	// RoleBindingActionGet is the action name to get a role binding
+	RoleBindingActionGet RoleBindingAction = "rolebinding_get"
+	// RoleBindingActionList is the action name to list role bindings
+	RoleBindingActionList RoleBindingAction = "rolebinding_list"
+)
+
+// ResourceRoleBindingV2 describes the relationships that will be created
+// for a resource to support role-binding V2
+type ResourceRoleBindingV2 struct {
+	// AvailableRolesFrom is the list of resource types that can provide roles to this resource
+	// Note that not all roles are available to all resources. This relationship is used to
+	// determine which roles are available to a resource.
+	// Before creating a role binding for a resource, one should check whether or
+	// not the role is available for the resource.
+	//
+	// Also see the RoleOwners field in the RBAC struct
+	AvailableRolesFrom []string
+}
```

Comment:
Three things:

* `ResourceRoleBindingV2` doesn't really convey what this is for, but I'm not sure what would
* `AvailableRolesFrom` as a name feels confusing, but I'm not sure what's better. I'll think about it
* The comment for `AvailableRolesFrom` is not correct here as far as I can tell. It's a list of relations, right?